benchmarks/benchmarks/benchmarks.py

```diff
@@ -71,25 +73,59 @@ def time_glm_hierarchical(self):
 
         n_counties = len(data.county.unique())
         with pm.Model():
-            # Hyperpriors for group nodes
             mu_a = pm.Normal('mu_a', mu=0., sd=100**2)
             sigma_a = pm.HalfCauchy('sigma_a', 5)
             mu_b = pm.Normal('mu_b', mu=0., sd=100**2)
             sigma_b = pm.HalfCauchy('sigma_b', 5)
-
-            # Intercept for each county, distributed around group mean mu_a
-            # Above we just set mu and sd to a fixed value while here we
-            # plug in a common group distribution for all a and b (which are
-            # vectors of length n_counties).
             a = pm.Normal('a', mu=mu_a, sd=sigma_a, shape=n_counties)
-            # Intercept for each county, distributed around group mean mu_a
             b = pm.Normal('b', mu=mu_b, sd=sigma_b, shape=n_counties)
+            eps = pm.HalfCauchy('eps', 5)
+            radon_est = a[county_idx] + b[county_idx] * data.floor.values
+            pm.Normal('radon_like', mu=radon_est, sd=eps, observed=data.log_radon)
+            pm.sample(draws=2000, njobs=4)
+
+
+class EffectiveSampleSizeSuite(object):
+    """Tests effective sample size per second on models
+    """
+    timeout = 360.0
+    params = (
+        [pm.NUTS, pm.Metropolis],  # Slice too slow, don't want to tune HMC
+        ['advi', 'jitter+adapt_diag', 'advi+adapt_diag_grad'],
+    )
+    param_names = ['step', 'init']
+
+    def setup(self, step, init):
+        """Initialize model and get start position"""
+        np.random.seed(123)
+        self.chains = 4
+        data = pd.read_csv(pm.get_data('radon.csv'))
+        data['log_radon'] = data['log_radon'].astype(theano.config.floatX)
+        county_idx = data.county_code.values
+        n_counties = len(data.county.unique())
+        with pm.Model() as self.model:
+            mu_a = pm.Normal('mu_a', mu=0., sd=100**2)
+            sigma_a = pm.HalfCauchy('sigma_a', 5)
+
+            mu_b = pm.Normal('mu_b', mu=0., sd=100**2)
+            sigma_b = pm.HalfCauchy('sigma_b', 5)
 
-            # Model error
+            a = pm.Normal('a', mu=mu_a, sd=sigma_a, shape=n_counties)
+            b = pm.Normal('b', mu=mu_b, sd=sigma_b, shape=n_counties)
             eps = pm.HalfCauchy('eps', 5)
 
             radon_est = a[county_idx] + b[county_idx] * data.floor.values
 
-            # Data likelihood
             pm.Normal('radon_like', mu=radon_est, sd=eps, observed=data.log_radon)
-            pm.sample(draws=2000, njobs=4)
+            self.start, _ = pm.init_nuts(chains=self.chains, init=init)
```

Comment:
This doesn't actually use the initialisation method specified. `init_nuts` doesn't only set the start value, but generates the step method, too. The behaviour of the mass matrix (including adaptation) is in the step method, and that is ignored here.

Reply:
ah shoot -- missed that.  i think this benchmark is trying to do too much, anyways.  i'll split this benchmark to have one comparing `NUTS` with different initializations, and have a different one that compares `Metropolis` with `NUTS` (with automatic initializations).  We sort of know what that second benchmark will look like, but I think it will be useful for talks and documentation to be able to point to how much more efficient Hamiltonian methods are, and for straightforward models.